plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainerInitializer.java

```diff
@@ -92,17 +93,33 @@ private static boolean jarPathsAreValid(LibraryClasspathContainer container) {
     for (IClasspathEntry classpathEntry : classpathEntries) {
       if (!classpathEntry.getPath().toFile().exists()
           || (classpathEntry.getSourceAttachmentPath() != null
-          && !classpathEntry.getSourceAttachmentPath().toFile().exists())) {
+              && !classpathEntry.getSourceAttachmentPath().toFile().exists())) {
         return false;
       }
     }
     return true;
   }
 
+  @Override
+  public boolean canUpdateClasspathContainer(IPath containerPath, IJavaProject project) {
+    return true;
+  }
+
+  @Override
+  public void requestClasspathContainerUpdate(
+      IPath containerPath, IJavaProject project, IClasspathContainer containerSuggestion)
+      throws CoreException {
+
+    LibraryClasspathContainerResolverJob job =
+        new LibraryClasspathContainerResolverJob(resolverService, project);
+    job.setUser(true);
+    job.setRule(BuildPath.resolvingRule(project));
```

Comment:
What do you think of setting this in the job constructor? That may be safer going forward.